File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/generate/Templates.java
package io.github.potjerodekool.openapi.common.generate;

import io.github.potjerodekool.codegen.template.adapter.EnumModelAdapter;

import org.stringtemplate.v4.ST;
import org.stringtemplate.v4.STGroupDir;

public class Templates {

    private final STGroupDir group = new STGroupDir("codegen-templates");

    public Templates() {
        group.registerModelAdaptor(Enum.class, new EnumModelAdapter());
    }

    public ST getInstanceOf(final String name) {
        return group.getInstanceOf(name);
    }
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/generate/Templates.kt
package io.github.potjerodekool.openapi.common.generate

import io.github.potjerodekool.codegen.template.adapter.EnumModelAdapter
import org.stringtemplate.v4.ST
import org.stringtemplate.v4.STGroupDir

class Templates {
    private val group = STGroupDir("codegen-templates")

    init {
        group.registerModelAdaptor(Enum::class.java, EnumModelAdapter())
    }

    fun getInstanceOf(name: String?): ST {
        return group.getInstanceOf(name)
    }
}
